src/ngResource/resource.js

```diff
@@ -625,6 +703,7 @@ angular.module('ngResource', ['ng']).
                 }
               }
 
+              value.$cancelRequest = cancelRequest;
```

Comment:
Not holding to the promise is taken care of in the finally block.

--
At this point `cancelRequest` will either be `angular.noop` or undefined.
The idea was to add `noop` back if it was there or set the property to undefined if it wasn't defined.

On a second thought, it would make more sense to avoid adding a property that didn't exist in the first place:

```js
if (cancelRequest) value.$cancelRequest = cancelRequest;
```